### PR TITLE
Refact/virtual display dummy display

### PR DESCRIPTION
**Pull request by fufesou**
https://github.com/rustdesk/rustdesk/issues/4359#issuecomment-1572621649

This PR is only partially tested, by setting `dummy_display_side_max_size` to 2000.

### CODE REVIEW

File: src/server/video_service.rs [hunk 1]
```diff
@@ -993,21 +986,38 @@ fn try_get_displays() -> ResultType<Vec<Display>> {
...
+#[cfg(windows)]
+fn no_displays(displays: &Vec<Display>) -> bool {
+    let display_len = displays.len();
+    if display_len == 0 {
+        true
+    } else if display_len == 1 {
+        let display = &displays[0];
+        let dummy_display_side_max_size = 800;
+        display.width() < dummy_display_side_max_size
+            && display.height() < dummy_display_side_max_size
```
Comment (rustdesk) [inline src/server/video_service.rs]: <=

**Comment by rustdesk** [comment]
you can test this by adjust your display to low resolution.

**Comment by rustdesk** [comment]
or test with a bigger dummy_display_side_max_size

**Comment by fufesou** [comment]
I've tested setting `dummy_display_side_max_size` to 1200.

New virtual display is plug in if the resolution is `1024x768` or `800x600`.

**Comment by fufesou** [comment]
There's a side effect that the resolution will be reset to original when plug in virtual display.

`dummy_display_side_max_size` is set to 1200.

I mannually set the remote resolution to `800x600`.
The resolution is set `2160x1440` automatically when the connection is established and the virtual display is plug in.
There's no record of resolution restore.

I'll try to find the reason.

**Comment by fufesou** [comment]
It works fine.

The resolution of display 0 is set to `2160x1440` when the virtual display is plug in.

Because the setting of two displays is set by previous operations.
When new monitor is plug in, the setting is applied.

**Pull request merged**